tests/integrationv2/bin/SSLSocketClient.java

```diff
@@ -0,0 +1,96 @@
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.security.KeyStore;
+import java.io.FileInputStream;
+import java.io.OutputStream;
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.SSLSocket;
+
+/*
+ * Simple JDK SSL client for integration testing purposes
+ */
+
+public class SSLSocketClient {
+    public static void main(String[] args) throws Exception {
+        int port = Integer.parseInt(args[0]);
+        String certificatePath = args[1];
+        String protocol = sslProtocols(args[2]);
+        String[] protocolList = new String[] {protocol};
+        String[] cipher = new String[] {args[3]};
+
+        String host = "localhost";
+        byte[] buffer = new byte[100];
+
+        SSLSocketFactory socketFactory = createSocketFactory(certificatePath, protocol);
+
+        try (
+            SSLSocket socket = (SSLSocket)socketFactory.createSocket(host, port);
+            OutputStream out = new BufferedOutputStream(socket.getOutputStream());
+            BufferedInputStream stdIn = new BufferedInputStream(System.in);
+        ) {
+            socket.setEnabledProtocols(protocolList);
+            socket.setEnabledCipherSuites(cipher);
+            socket.startHandshake();
+            System.out.println("Starting handshake");
+
+            while (stdIn.read(buffer) != -1) {
+                out.write(buffer);
+            }
+            out.flush();
+
+            out.close();
+            socket.close();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```

Comment:
Where does the `return 1` happen? This looks like it should exit cleanly.

Reply:
A java void method should not return anything.

Reply:
Spoke offline, never mind this comment, your proposed idea about alerts sounds good.